sphinx/search/__init__.py

```diff
@@ -242,6 +243,7 @@ def __init__(self, env: BuildEnvironment, lang: str, options: Dict, scoring: str
         # stemmed words in titles -> set(docname)
         self._title_mapping: Dict[str, Set[str]] = {}
         self._all_titles: Dict[str, List[Tuple[str, str]]] = {}  # docname -> all titles
+        self._index_entries: Dict[str, List[Tuple[str, str, str]]] = {}  # docname -> index entry
```

Comment:
i feel like this section is crying out for some type aliases to simplify the signatures here